src/librustc_typeck/diagnostics.rs

```diff
@@ -265,6 +323,28 @@ RFC. It is, however, [currently unimplemented][iss15872].
 [iss15872]: https://github.com/rust-lang/rust/issues/15872
 "##,
 
+E0121: r##"
+When `_` is in a type, it can act as a placeholder for another type. The
```

Comment:
I find it somewhat unusual that the error this code is referencing isn't mentioned first (omitting the type of a signature that's required) versus explaining what `_` does. Perhaps this could be restructured first to stay consistent with the other error descriptions found here as well?